test/OpenTelemetry.Tests.Stress/Meat.cs

```diff
@@ -9,11 +9,19 @@ public partial class Program
 {
     public static void Main()
     {
-        Stress(concurrency: 1, prometheusPort: 9464);
+        RunStressTest(new()
+        {
+            Concurrency = 1,
+            PrometheusInternalMetricsPort = 9464,
+        });
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     protected static void Run()
     {
     }
+
+    protected static void WriteRunInformationToConsole(StressTestOptions options)
+    {
+    }
```

Comment:
Is there a way to get rid of this extra burden?

Reply:
I went ahead and refactored things. I was trying to keep this small but the way things were set up was a bit strange IMO and causing stuff like this to be forced downstream. There is now some class hierarchy and the projects reference the base test assembly instead of including the files directly (and mirroring the dependencies in their projects).

Reply:
I personally prefer simple/small things, but no strong opinion and don't want to block this PR.
Could you @CodeBlanch verify the perf of an empty test (which means `Run` is an empty function)? I want to make sure that the CPU cycles spent on each loop remains the same (in other words, the hierarchy is not making things slower) - this is important since we use stress test for some high perf code, and I want to make sure our overhead is small (I mean something like 3 CPU instructions per loop).

Reply:
When I run the demo one as it is today (c=1) it gives me Average CPU Cycles/Loop: 0. When I run the demo one as it is with this PR with c=1 it gives me Average CPU Cycles/Loop: 1. Pretty low overhead!

When I start cranking up the concurrency I start to see some overhead for old and new. This helps that stabilize: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5381/commits/24e587b91a70c2379a6a06ddfaa4b1ea4c6eca05

Reply:
Awesome! Thanks @CodeBlanch!